### PR TITLE
Fix Windows CUDA periodic inductor/test_pattern_matcher test

### PR DESCRIPTION
**Pull request by huydhn**
The check was run on Windows and crashed there because Windows doesn't have triton, i.e. https://github.com/pytorch/pytorch/actions/runs/9606662121/job/26502347998#step:15:13196

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: test/inductor/test_pattern_matcher.py [hunk 1]
```diff
@@ -24,17 +24,19 @@
...
 from torch.testing._internal.common_cuda import SM80OrLater
 from torch.testing._internal.common_utils import IS_LINUX, LazyVal, skipIfRocm
 from torch.testing._internal.inductor_utils import HAS_CUDA
 from torch.utils import _pytree as pytree
 
-
-is_A100 = LazyVal(
-    lambda: torch.cuda.is_available() and get_gpu_shared_memory() == 166912
+# NVIDIA A100-SXM4-40GB
+is_a100 = LazyVal(
```
Comment (ZainRizvi) [inline test/inductor/test_pattern_matcher.py:36]: What do you think about renaming `is_a100` to `is_a100_linux` or something else taht more accurately reflects the constraints of the test?

**Review by ZainRizvi — APPROVED** [review]

File: test/inductor/test_pattern_matcher.py (hunk 1 above)
Comment (huydhn) [reply test/inductor/test_pattern_matcher.py:36]: Sure, that seems like an easy tweak

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**